integrations/kube-agent-updater/pkg/controller/status_writer.go

```diff
@@ -0,0 +1,132 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package controller
+
+import (
+	"context"
+	"path/filepath"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/gravitational/teleport/lib/autoupdate/agent"
+)
+
+const updaterIDFile = agent.BinaryName + ".id"
+
+type StatusWriter struct {
+	kclient.Client
+	// Scheme for Kubernetes objects.
+	Scheme *runtime.Scheme
+	// UpdateID is the unique ID of the updater.
+	UpdateID uuid.UUID
+	// UpdateGroup is the group of the updater.
+	UpdateGroup string
+	// ProxyAddress is the address of the proxy.
+	ProxyAddress string
+}
+
+// writeStatus updates the configmap with the latest status of the updater.
+func (c *StatusWriter) writeStatus(ctx context.Context, owner metav1.Object, version string, failed bool) error {
```

Comment:
Should we make this a patch from the existing status? As the client is cached, this would make most cycles a noop. I fear that the updater will do _a lot_ of writes with this change.

Reply:
`CreateOrUpdate` compares first and won't hit the API server at all unless there are changes. Additionally, we only change the timestamp on failure.